src/main/kotlin/it/pagopa/wallet/services/PaymentMethodsService.kt

```diff
@@ -0,0 +1,44 @@
+package it.pagopa.wallet.services
+
+import it.pagopa.generated.ecommerce.model.PaymentMethodResponse
+import it.pagopa.wallet.client.EcommercePaymentMethodsClient
+import it.pagopa.wallet.repositories.PaymentMethodsTemplateWrapper
+import kotlinx.coroutines.reactor.mono
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.switchIfEmpty
+
+@Service
+class PaymentMethodsService(
+    @Autowired private val paymentMethodsRedisTemplate: PaymentMethodsTemplateWrapper,
+    @Autowired private val ecommercePaymentMethodsClient: EcommercePaymentMethodsClient,
+) {
+
+    /*
+     * Logger instance
+     */
+    var logger: Logger = LoggerFactory.getLogger(this.javaClass)
+
+    fun getPaymentMethodById(paymentMethodId: String): Mono<PaymentMethodResponse> =
+        mono {
+                logger.debug("Try to retrieve payment method from cache: [$paymentMethodId]")
+                paymentMethodsRedisTemplate.findById(paymentMethodId)
+            }
+            .doOnNext { logger.info("Cache hit for payment method with id: [$paymentMethodId]") }
+            .switchIfEmpty {
+                logger.info("Cache miss for payment method: [$paymentMethodId]")
+                retrievePaymentMethodById(paymentMethodId)
+            }
+
+    private fun retrievePaymentMethodById(paymentMethodId: String): Mono<PaymentMethodResponse> =
+        ecommercePaymentMethodsClient
+            .getPaymentMethodById(paymentMethodId)
+            .doOnSuccess {
+                logger.debug("Save payment method into cache: [$paymentMethodId]")
+                paymentMethodsRedisTemplate.save(it)
```

Comment:
In this method, the doOnError assumes only an error when calling payment method, but this doOnError can also be triggered by an error on paymentMethodsRedisTemplate.save(it).

Maybe we need to add a specific error logging for paymentMethodsRedisTemplate.save(it) to differentiate the 2 cases?

Something like this, just an idea

```  
            .doOnSuccess { paymentMethodResponse ->
                logger.debug("Save payment method into cache: [$paymentMethodId]")
                paymentMethodsCache.save(paymentMethodResponse)
                    .doOnError { ex ->
                        logger.error("Error saving payment method to cache: [$paymentMethodId]", ex)
                    }
                    .subscribe()
            }
`

Reply:
Current redisTemplate implementation is not reactive, so i can't use doOnError and subscribe methods.

Reply:
Then why not use a non-reactive error handling?

Just an idea
 ```               try {
                    paymentMethodsRedisTemplate.save(it)
                } catch (e: Exception) {
                    logger.error("Error saving payment method to cache: [$paymentMethodId]", e)
                }
                `

Reply:
Done here 22742f2cc55183020ac96105bfe640cf19337c50